x/concentrated-liquidity/incentives_test.go

```diff
@@ -401,6 +435,84 @@ func (s *KeeperTestSuite) TestCalcAccruedIncentivesForAccum() {
+				expectedIncentives(incentiveRecordOneWithDifferentStartTime.IncentiveDenom, incentiveRecordOne.EmissionRate, time.Hour, sdk.NewDec(100)), // since we have 2 records with same denom, the rate of emission went up x2
+				expectedIncentives(incentiveRecordOneWithDifferentDenom.IncentiveDenom, incentiveRecordOne.EmissionRate, time.Hour, sdk.NewDec(100)),     // since we have 2 records with same denom, the rate of emission went up x2
+			},
+			expectedIncentiveRecords: []types.IncentiveRecord{
+				// We only charge the first incentive record because the second minUpTime hasn't been hit yet
+				chargeIncentive(incentiveRecordOneWithDifferentStartTime, time.Hour),
+				chargeIncentive(incentiveRecordOneWithDifferentDenom, time.Hour),
+			},
+			expectedPass: true,
+		},
+		"two incentive records with same denom, different min up-time": {
+			poolId:              defaultPoolId,
+			accumUptime:         types.SupportedUptimes[0],
+			qualifyingLiquidity: sdk.NewDec(100),
+			timeElapsed:         time.Hour,
+
+			poolIncentiveRecords: []types.IncentiveRecord{incentiveRecordOne, incentiveRecordOneWithDifferentMinUpTime},
+
+			expectedResult: sdk.DecCoins{
+				// We expect both incentive record to qualify
+				expectedIncentives(incentiveRecordOne.IncentiveDenom, incentiveRecordOne.EmissionRate, time.Hour, sdk.NewDec(100)), // since we have 2 records with same denom, the rate of emission went up x2
+			},
+			expectedIncentiveRecords: []types.IncentiveRecord{
+				// We only charge the first incentive record because the second minUpTime hasn't been hit yet
+				chargeIncentive(incentiveRecordOne, time.Hour),
+				incentiveRecordOneWithDifferentMinUpTime,
+			},
+			expectedPass: true,
+		},
+		"two incentive records with  same accum uptime and start time across multiple records with different denoms": {
+			poolId:              defaultPoolId,
+			accumUptime:         types.SupportedUptimes[0],
+			qualifyingLiquidity: sdk.NewDec(100),
+			timeElapsed:         time.Hour,
+
+			poolIncentiveRecords: []types.IncentiveRecord{incentiveRecordOne, incentiveRecordOneWithDifferentDenom},
+
+			expectedResult: sdk.DecCoins{
+				// We expect both incentive record to qualify
+				expectedIncentives(incentiveRecordOne.IncentiveDenom, incentiveRecordOne.EmissionRate, time.Hour, sdk.NewDec(100)),                   // since we have 2 records with same denom, the rate of emission went up x2
+				expectedIncentives(incentiveRecordOneWithDifferentDenom.IncentiveDenom, incentiveRecordOne.EmissionRate, time.Hour, sdk.NewDec(100)), // since we have 2 records with same denom, the rate of emission went up x2
+			},
+			expectedIncentiveRecords: []types.IncentiveRecord{
+				// We only charge the first incentive record because the second minUpTime hasn't been hit yet
+				chargeIncentive(incentiveRecordOne, time.Hour),
+				chargeIncentive(incentiveRecordOneWithDifferentDenom, time.Hour),
+			},
+			expectedPass: true,
+		},
```

Comment:
In the issue, there was also a test for:
- Two denoms each has 2 start times and 2 uptimes where there is the same accum uptime and start time across multiple records with different denoms

what this implies is having at least 4 records where only a subset of them is eligible for emitting incentives

Reply:
Added here(https://github.com/osmosis-labs/osmosis/pull/4353/commits/53db2efba9a9598f3daec6649438a8325d4dd269#diff-91569dd5b56ddbfc9450f57de47bc13061e1daceaba154cab15b635d98912d19R514) 

lmk if this is what you meant, the wording got me a lil confused